flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/reconciler/JobReconcilerTest.java

```diff
@@ -143,6 +144,10 @@ public void testUpgradeModeChangeFromSavepointToLastState() throws Exception {
         deployment.getSpec().getJob().setState(JobState.RUNNING);
         deployment.getSpec().setImage("new-image-2");
 
+        // Wait ready for spec changes
+        deployment.getStatus().setJobManagerDeploymentStatus(JobManagerDeploymentStatus.READY);
+        deployment.getStatus().getJobStatus().setState(JobState.RUNNING.name());
+
```

Comment:
This should not be necessary, in this test we simulate the SUSPENDED -> RUNNING

Reply:
good point.